ext/bg/js/search-query-parser.js

```diff
@@ -17,73 +17,47 @@
  */
 
 
-class QueryParser {
+class QueryParser extends TextScanner {
     constructor(search) {
+        super(document.querySelector('#query-parser'), [], [], []);
         this.search = search;
-        this.pendingLookup = false;
-        this.clickScanPrevent = false;
 
         this.parseResults = [];
         this.selectedParser = null;
 
         this.queryParser = document.querySelector('#query-parser');
```

Comment:
ESLint doesn't like setting `this.queryParser` before `super()` (`no-this-before-super`), so `document.querySelector` is used twice. It could be avoided by creating a new local constant, but that would make it more complex.